frontend/src/reducers/queries.ts

```diff
@@ -0,0 +1,216 @@
+import { IQuery } from '../common/models/apps/querys';
+import { Routine } from 'redux-saga-routines';
+import {
+  duplicateSelectQueryRoutine,
+  fetchQueryRoutine,
+  openQueryRoutine,
+  setNewCodeRoutine,
+  setNewConfirmRoutine,
+  setNewNameQueryRoutine,
+  setNewRunRoutine,
+  setSelectQueryRoutine, setSuccessTriggersRoutine,
+  setUnSuccessTriggersRoutine,
+  setWaiterQueryRoutine
+} from '../scenes/constructor/routines';
+import { ITrigger } from '../common/models/query/ITrigger';
+
+export interface IQueryState {
+  queriesApp: Array<IQuery>;
+  selectQuery:{
+    selectQueryId: string,
+    selectQueryName: string,
+    selectQueryCode: string,
+    selectQueryTriggers:Array<ITrigger>|[],
+    runAutomatically: boolean,
+    showConfirm: boolean
+  },
+  setNewCode:string,
+  setNewRun: boolean,
+  runAutomaticallyTitle:string,
+  setNewName:string,
+  setNewConfirm:boolean,
+  setNewSuccessTriggers:Array<ITrigger>|[],
+  setNewUnSuccessTriggers:Array<ITrigger>|[],
+  queriesAppLength:number,
+  isLoading: boolean,
+  isOpen:boolean,
+  isDuplicate:boolean,
+  waitingQuery:{
+    QueryId?:string,
+    QueryName?:string,
+    QueryCode?:string,
+    QueryTriggers:Array<ITrigger>|[],
+    runAutomatically: boolean,
+    showConfirm: boolean
+  }
+}
+
+const initialState:IQueryState = {
+  queriesApp: [],
+  selectQuery: {
+    selectQueryId: '',
+    selectQueryName: '',
+    selectQueryCode: '',
+    selectQueryTriggers: [],
+    runAutomatically: false,
+    showConfirm: false
+  },
+  setNewCode: '',
+  runAutomaticallyTitle: 'Run query automatically when inputs change',
+  setNewName: '',
+  setNewRun: false,
+  setNewConfirm: false,
+  setNewSuccessTriggers: [],
+  setNewUnSuccessTriggers: [],
+  queriesAppLength: 0,
+  isOpen: false,
+  isDuplicate: false,
+  waitingQuery: {
+    QueryId: '',
+    QueryName: '',
+    QueryCode: '',
+    QueryTriggers: [],
+    runAutomatically: false,
+    showConfirm: false
+  },
+  isLoading: true
+};
+
+export const queries = (state = initialState, action: Routine<any>): IQueryState => {
+  let successTriggers:Array<ITrigger>|[] = [];
+  let UnSuccessTriggers:Array<ITrigger>|[] = [];
+  switch (action.type) {
+    case fetchQueryRoutine.SUCCESS:
+      return {
+        ...state,
+        queriesApp: [...action.payload],
+        queriesAppLength: action.payload.length,
+        isLoading: false
+      };
+    case openQueryRoutine.SUCCESS:
+      const runTitle = action.payload[0].runAutomatically ? 'Run query only when manually triggered'
+        : 'Run query automatically when inputs change';
+      action.payload[0].triggers.find((element:ITrigger) => {
```

Comment:
Why do you need to use `find` here? I'm not sure I understand what you're trying to do, but seems like you need some other method, like `forEach`